IPython/core/display.py

```diff
@@ -1338,3 +1448,15 @@ def set_matplotlib_close(close=True):
     from ipykernel.pylab.config import InlineBackend
     cfg = InlineBackend.instance()
     cfg.close_figures = close
+
+
+
+for oname in __all__:
+    # set the module name to IPython.display or sphinx cannot correctly cross
+    # link. Mostly it figures out that objets are in IPython.core.display, but
+    # that's a module we explicitly do not document.
+    try:
+        # using hasattr to check still raises an AttributeError.
+        locals()[oname].__module__ = 'IPython.display' 
```

Comment:
Functions defined in one module can be referenced in any module in which they are importable, we just have to make sure that they are present in the autodoc module for the desired target.

I do this [in pyzmq](https://github.com/zeromq/pyzmq/blob/master/docs/source/api/zmq.rst), where docs treat everything as being in top-level `zmq` module, but classes are actually defined all over the place.